snooty/postprocess.py

```diff
@@ -976,7 +983,63 @@ def enter_node(self, fileid_stack: FileIdStack, node: n.Node) -> None:
             assert isinstance(argument, n.Reference)
             source = argument.refuri
 
-        self.openapi_pages[current_slug] = self.SourceData(source_type, source)
+        api_version = node.options.get("api-version", None)
+        resource_versions: Optional[List[str]] = None
+
+        # Fetch OpenAPI versioning data if options are present
+        if api_version and source == "cloud":
+            # Fetch latest git_hash for S3 versioning data
+            try:
+                git_hash_url = "https://cloud.mongodb.com/version"
+                git_hash_response = util.HTTPCache.singleton().get(git_hash_url)
+                git_hash = str(git_hash_response, "utf-8")
+
+                version_url = f"https://mongodb-mms-prod-build-server.s3.amazonaws.com/openapi/{git_hash}-api-versions.json"
+                version_response = util.HTTPCache.singleton().get(version_url)
+                decoded = str(version_response, "utf-8")
+                data = yaml.safe_load(decoded)
```

Comment:
This might be a good place to use flutter!

```
@dataclass
@checked
class OpenAPIData:
    versions: Dict[str, List[str]]


data = check_type(OpenAPIData, yaml.safe_load(decoded))
```

This will take care of all of the type validation logic you currently have